external/gtest/CMakeLists.txt

```diff
@@ -1,30 +1,43 @@
-configure_file(CMakeLists.txt.in CMakeLists.txt @ONLY)
+cmake_minimum_required(VERSION 3.10)
+project(GoogleTest CXX)
 
-configure_file(CMakeLists.txt.in CMakeLists.txt)
-execute_process(COMMAND "${CMAKE_COMMAND}" -G "${CMAKE_GENERATOR}" .
-  RESULT_VARIABLE result
-  WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}"
-)
-if(result)
-  message(FATAL_ERROR "CMake step for googletest failed: ${result}")
+if(WIN32)
+    set(gtest_cxx_flags -D_UNICODE -DUNICODE -DSTRICT -DWIN32_LEAN_AND_MEAN)
+else()
+    set(gtest_cxx_flags -Wno-unused-parameter -Wno-missing-field-initializers)
```

Comment:
`-Wno-unused-parameter`
`-Wno-missing-field-initializers`
1. What do they do?
2. What platform?

Reply:
2. GCC on Linux/macOS + Clang on macOS

1. See https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html
-Wunused-parameter
-Wmissing-field-initializers

-W**no**-... - suppress these warnings.

Same applies to Clang.